app/models/order.rb

```diff
@@ -47,6 +64,10 @@ def set_initial_state
     before_transition on: :submit do |order|
       order.add_to_stockit
     end
+
+    after_transition on: :submit do |order|
+      order.update_packages
```

Comment:
I would put the guard on this line, so

    order.update_packages if (detail_type == "GoodCity")

That way, the function `update_packages` (renamed to `designate_orders_packages`) would do exactly that in every case it is called. It would be surprising to use the function later on another detail_type and find that it doesn't designate so this would fix that case